docs/fhir-guide/FHIR 101 - Practical Guide.ipynb

```diff
@@ -0,0 +1,2966 @@
+    "\n",
+    "📘Learn more about Kids First [IG generation and model validation](https://github.com/kids-first/kf-model-fhir/blob/master/README.md) using IG Publisher "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true,
+    "hidden": true
+   },
+   "source": [
+    "### Model Validation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "There are 2 primary ways to validate:\n",
+    "\n",
+    "1. Use the [HL7 IG Publisher](https://confluence.hl7.org/display/FHIR/IG+Publisher+Documentation) Java CLI\n",
+    "2. Use a FHIR server\n",
+    "    - POST StructureDefinitions to `/StructureDefinition` endpoint\n",
+    "    - POST example Resources to `<resourceType>/$validate` endpoint\n",
+    "\n",
+    "The most thorough and standard way to validate a FHIR data model is to use the IG Publisher. The IG Publisher is under active development and is the most up to date in terms of its validation logic. However, since the IG Publisher tightly couples IG site generation with validation it can be complicated to use. \n",
+    "\n",
+    "The second method is easier but not standard. All compliant FHIR servers will implement basic validation logic, however, some servers may or may not implement additional validation/checks that the IG Publisher does.\n",
+    "\n",
+    "To keep things simple, we will use the second method of validation."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "⚡️**Key Take Away** - the most thorough way to validate a FHIR data model is using the HL7 IG Publisher"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true
+   },
+   "source": [
+    "## Server Development"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true,
+    "hidden": true
+   },
+   "source": [
+    "### Pick an Existing Solution\n",
+    "You might have noticed that there are not many open source or free FHIR servers currently in development. This is likely due to a couple of factors:\n",
+    "\n",
+    "- The FHIR specification is rapidly evolving and the tooling around it is still in early stages of development. \n",
+    "- Server development is hard \n",
+    "\n",
+    "In order to develop a compliant FHIR server you must: \n",
+    "- Implement the FHIR model validation logic. This is not easy since a FHIR model is authored  \n",
+    "in the FHIR DSL. The server must know how to parse the DSL (a.k.a Conformance Resources) and\n",
+    "execute rules defined in them.\n",
+    "- Implement a parser and query constructor for the FHIR RESTful query language - which is also custom/not based on any known standard. Additionally, the kinds of search queries that can be made with a FHIR search API are fairly rich and complex.\n",
+    "\n",
```

Comment:
```suggestion
    "\n",
    "For these reasons it is best to select one of the existing FHIR server solutions rather than develop your own.\n",
    "\n",
```